graftm/hmmer.py

```diff
@@ -675,88 +693,83 @@ def d_search(self, files, args, run_stats, base, unpack, raw_reads, euk_check):
             The reads to be searched.
         euk_check : bool
             True False, whether to check the entire sample for euk reads.
-        
+        search_method : str
+            The method for searching e.g. 'hmmsearch' or 'diamond'
+        gpkg : obj
+            GraftM package object, created by graftm_package.py
+            
         Returns
         -------
         hit_reads : str
             The output fasta file of reads that hit 
-        run_stats : dict
-            Updated run stats for the file being worked on (inluding run times 
-            for each search and numbers of hits etc)
+        '''
+        
+        # Define outputs
+        hmmsearch_output_table = files.hmmsearch_output_path(base)
+        hit_reads_fasta        = files.fa_output_path(base)
+        
+        if search_method == "hmmsearch":
+            # First search the reads using the HMM
+            search_result, table_list = self.nhmmer(
+                                                    hmmsearch_output_table,
+                                                    unpack,
+                                                    args.threads,
+                                                    args.eval
+                                                    )
+            for result in search_result:
+                hit_readnames=[x[0] for x in result.each([SequenceSearchResult.QUERY_ID_FIELD])]
             
-        Raises
-        ------
-        N/A
+        elif search_method == 'diamond':
+            logging.error("Diamond searches not supported for nucelotide sequences yet")
```

Comment:
typo. Also why log an error if you are raising an exception anyway? So just delete this line. Also unclear whether you mean nucleotide query or database